src/nvim/api/vim.c

```diff
@@ -107,11 +108,19 @@ String nvim_exec(String src, Boolean output, Error *err)
   if (output) {
     msg_silent++;
   }
+
+  const sctx_T save_current_sctx = current_sctx;
+  current_sctx.sc_sid =
```

Comment:
`current_sctx` should not be changed for `VIML_INTERNAL_CALL`. probably could break it out to a helper function.

Reply:
done